collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDownloader.java

```diff
@@ -141,10 +142,16 @@ public boolean processOneForm(ServerFormDetails fd, FormDownloaderListener state
                 fileResult = downloadXform(fd.getFormName(), fd.getDownloadUrl(), stateListener, tempDir, formsDirPath);
 
                 if (fd.getManifest() != null) {
-                    finalMediaPath = FileUtils.constructMediaPath(
-                            fileResult.getFile().getAbsolutePath());
-                    String error = downloadManifestAndMediaFiles(tempMediaPath, finalMediaPath, fd,
-                            stateListener, fd.getManifest().getMediaFiles(), tempDir);
+                    finalMediaPath = FileUtils.constructMediaPath(formsDirPath + File.separator + fileResult.file.getName());
+                    String error = downloadManifestAndMediaFiles(
+                            tempMediaPath,
+                            finalMediaPath,
```

Comment:
Is this an example of where we might be able to have a `saveMediaFile` method on the `FormsRepository` that takes in a media filename, form_id, and file contents instead of building the path here? Then `downloadManifestAndMediaFiles` can take a `FormsRepository` instead.

Reply:
Yeah I think so! I'm thinking `downloadManifestAndMediaFiles` would be a lot simpler if it used the repositories.